tsl/test/isolation/specs/compression_chunk_race.spec.in

```diff
@@ -0,0 +1,102 @@
+###
+
+setup {
+   CREATE OR REPLACE FUNCTION debug_waitpoint_enable(TEXT) RETURNS VOID LANGUAGE C VOLATILE STRICT
+     AS '@TS_MODULE_PATHNAME@', 'ts_debug_point_enable';
+
+   CREATE OR REPLACE FUNCTION debug_waitpoint_release(TEXT) RETURNS VOID LANGUAGE C VOLATILE STRICT
+     AS '@TS_MODULE_PATHNAME@', 'ts_debug_point_release';
+
+   CREATE TABLE sensor_data (
+   time timestamptz not null,
+   sensor_id integer not null,
+   cpu double precision null,
+   temperature double precision null);
+
+   -- Create large chunks that take a long time to compress
+   SELECT FROM create_hypertable('sensor_data','time', chunk_time_interval => INTERVAL '14 days');
+
+   INSERT INTO sensor_data
+   SELECT
+   time + (INTERVAL '1 minute' * random()) AS time,
+   sensor_id,
+   random() AS cpu,
+   random()* 100 AS temperature
+   FROM
+   generate_series('2022-01-01', '2022-01-15', INTERVAL '1 minute') AS g1(time),
+   generate_series(1, 50, 1) AS g2(sensor_id)
+   ORDER BY time;
+
+   SELECT count(*) FROM sensor_data;
+   
+   ALTER TABLE sensor_data SET (timescaledb.compress, timescaledb.compress_segmentby = 'sensor_id, cpu');
+}
+
+teardown {
+   DROP TABLE sensor_data;
+}
+
+
+session "s1"
+setup {
+   SET client_min_messages=ERROR; -- Suppress chunk "_hyper_XXX_chunk" is not compressed messages 
+}
+
+step "s1_compress" {
+   SELECT compression_status FROM chunk_compression_stats('sensor_data');
+   SELECT 1 FROM (SELECT compress_chunk(i, if_not_compressed => true) FROM show_chunks('sensor_data') i) i;
+   SELECT compression_status FROM chunk_compression_stats('sensor_data');
+   SELECT count(*) FROM sensor_data;
+}
+
+step "s1_decompress" {
+   SELECT 1 FROM (SELECT decompress_chunk(i, if_compressed => true) FROM show_chunks('sensor_data') i) i;
+   SELECT compression_status FROM chunk_compression_stats('sensor_data');
+   SELECT count(*) FROM sensor_data;
+}
+
+session "s2"
+setup {
+   SET client_min_messages=ERROR; -- Suppress chunk "_hyper_XXX_chunk" is not compressed messages 
+}
+
+step "s2_compress" {
+   SELECT compression_status FROM chunk_compression_stats('sensor_data');
+   SELECT 1 FROM (SELECT compress_chunk(i, if_not_compressed => true) FROM show_chunks('sensor_data') i) i;
+   SELECT compression_status FROM chunk_compression_stats('sensor_data');
+   SELECT count(*) FROM sensor_data;
+   RESET client_min_messages;
+}
+
+step "s2_decompress" {
+   SELECT 1 FROM (SELECT decompress_chunk(i, if_compressed => true) FROM show_chunks('sensor_data') i) i;
+   SELECT compression_status FROM chunk_compression_stats('sensor_data');
+   SELECT count(*) FROM sensor_data;
+   RESET client_min_messages;
+}
+
+session "s3"
+step "s3_lock_compression" {
+    SELECT debug_waitpoint_enable('compress_chunk_impl_start');
+}
+
+step "s3_lock_decompression" {
+    SELECT debug_waitpoint_enable('decompress_chunk_impl_start');
+}
+
+step "s3_unlock_compression" {
+    SELECT debug_waitpoint_release('compress_chunk_impl_start');
+}
+
```

Comment:
It is theoretically possible that you serialize the sessions on the debug waitpoint (since they execute the same code) which would not show up in this test. You can solve this by adding a printout of the relevant locks before you actually release them, to catch bugs, e.g., using a `SELECT` from `pg_locks` in a separate step for the `s3` session.

Reply:
As discussed, a filtered output of `pg_locks` is added to the test.